src/Divider.js

```diff
@@ -10,7 +10,7 @@ import { colors, spacers } from './theme.js'
  * @returns {React.Component}
  * @example import { Divider } from @dhis2/ui-core
  */
-const Divider = ({ margin, className, dataTest }) => (
```

Comment:
Removing the margin prop is technically a breaking change. Could we simply deprecate it for now and remove it in the next major that is coming up?

E.g.:

```
 * @prop {string} [margin="${spacers.dp8} 0"] - DEPRECATED: A CSS shorthand declaration for margin
```

Reply:
Yes, but is there a way I can bypass the `no-unused-vars` lint error? I was thinking that if the prop is deprecated I could simply not use it, or am I mistaken?

In theory, I could keep the default value for the `margin` prop, but override it with the new `dense` prop. E.g. on line 25 of `Divider.js`:
`div {margin: ${dense ? spacers.dp4 : margin} 0;`?

Reply:
Not using it would also be a breaking change, so that option is a no-go. We need to support margin _and_ dense in a transition period.

The problem with this `margin: ${dense ? spacers.dp4 : margin} 0;` is that the `margin` prop accepts a short-hand CSS declaration string, i.e. this is valid for `margin`: `margin="10px 5px 0px 15px"` but that would be wrong in the example you gave.

It would need to be more along the lines of:

```
margin: ${ dense ? `${ spacers.dp4 } 0` : margin };
```

This means that `dense` and `margin` are mutually exclusive though, so that propType validation needs to be added too.

Reply:
Ah yes, I understand. The only issue I see now is that, looking at [@dhis2/prop-types](https://github.com/dhis2/prop-types#mutuallyexclusiveexlusivepropnames-proptype--error--null), it is not possible to set multiple propTypes.

dense is `bool` and margin is `string`. It doesn't seem to be possible to do:
```
propTypes.mutuallyExclusive(
    ['margin', 'dense'],
    [dense: propTypes.bool, margin: propTypes.string]
)
```
This is not a critical change, so if this causes issues then it can be left out until the more thorough rework on the menu.

Reply:
Haha, I did not think about that. Yeah, that probably won't fly.

OK -- then we can forgo the `mutuallyExclusive` propType validation and note it in the jsdoc string:

`* @prop {string} [margin="${spacers.dp8} 0"] - DEPRECATED: A CSS shorthand declaration for margin. If dense is used, it has precedence.`

Reply:
Cool 👍 I've made those changes.

Reply:
I actually think you could get this to work in the current form of `mutuallyExclusive`:
```javascript
const denseNotWithMargin = propTypes.mutuallyExclusive(
    ['margin', 'dense'],
    propTypes.string
)

const marginNotWithDense = propTypes.mutuallyExclusive(
    ['margin', 'dense'],
    propTypes.bool
)
```

I fully agree this is awkward 😬 
But it might be able to help you our here....